app/javascript/packs/shop.js

```diff
@@ -8,66 +11,65 @@ document.addEventListener("turbolinks:load", function () {
   const addToBasketButtons = document.querySelectorAll(".basket-add");
 
   addToBasketButtons.forEach((element) => {
-    element.addEventListener("click", () => {
-      console.log("click!");
-
-      const price = parseFloat(
-        element.attributes.getNamedItem("data-price").nodeValue
+    element.addEventListener("click", (event) => {
+      event.preventDefault();
+      const product = JSON.parse(
+        element.attributes.getNamedItem("data-product").value
       );
-      const name = element.form.elements.name.value;
-      const sku = element.form.elements.sku.value;
-      const onSale = element.form.elements.on_sale.value;
-      const originalPrice = element.form.elements.original_price.value;
 
-      updateTotal(price);
+      updateTotal(product.price);
       updateCount(1);
-      updateContents(name);
-      basketEntities.push(productEntityData(name, sku, onSale, originalPrice));
-      console.log(basketEntities);
+      updateBasketEntities(product);
+      updateContents();
 
-      // Tracking a product being added to the basket
+      // Tracking a product being added to the basket.
       window.snowplow("trackSelfDescribingEvent", {
         event: {
           schema: "iglu:test.example.iglu/basket_action_event/jsonschema/1-0-0",
           data: {
             action: "add",
           },
         },
-        // The specific product is included as an Entity in the Event context
+        // The specific product is included as an Entity in the Event context.
         context: [
           {
-            schema: "iglu:test.example.iglu/product_entity/jsonschema/1-0-0",
-            data: {
-              sku: sku,
-              name: name,
-              price: parseFloat(price),
-              onSale: onSale === "true",
-              startPrice: parseFloat(originalPrice),
-              quantity: 1,
-            },
+            schema: "iglu:test.example.iglu/product_entity/jsonschema/1-0-1",
```

Comment:
I did it like this to give an example of versioning. How often do people increase their schema versions? Happy to make into one schema if you think the example isn't useful

Reply:
People likely bump their schemas as their website evolves. Its fine to leave it in, it might make people think a little more about their schema evolution when they see this. The 1-0-0 to 1-0-1 is also a good example of an additive change to a schema so does serve as a good example.

Reply:
I'll leave it like this then